### ISSUE TITLE
`UnexpectedTracerError` when jitting simulator and using using a loss function afterwards

### ISSUE THREAD

**Issue by michaeldeistler** [jaxleyverse/jaxley#74]
The following gives an `UnexpectedTracerError` (full error message in comment):
```python
def simulate(mem_params, image):
    new_current = jnp.zeros((199, 784))
    new_current = new_current.at[80:, :].set(i_amp / 255 * image * jnp.ones((784,)))
    stims = stimuli.set_currents(new_current)

    s = solve(
        network,
        init_v,
        mem_states=mem_states,
        mem_params=mem_params,
        mem_channels=mem_channels,
        syn_states=syn_states,
        syn_params=syn_params,
        syn_channels=syn_channels,
        stimuli=stims,
        recordings=recs,
        checkpoint_lengths=None,
    )
    return s

jitted_sim = jit(vmap(simulate, in_axes=(None, 0)))
traces = jitted_sim(mem_params, train_images[:20])

def loss(mem_params, image, label):
    s = simulate(mem_params, image)
    return jnp.abs(s[0, -1] - label)

jitted_loss = vmap(loss, in_axes=(None, 0, 0))  # jit(vmap(...)) also gives error.
l = jitted_loss(mem_params, train_images[:20], train_labels[:20])
```

Crucially, the error gets resolved when I do not `jit` the `simulate` call:
```python
jitted_sim = vmap(simulate, in_axes=(None, 0))
traces = jitted_sim(mem_params, train_images[:20])
```
